batchglm/models/glm_norm/utils.py

```diff
@@ -71,3 +42,74 @@ def compute_scales_fun(variance, mean):
         link_fn=link_fn,
         compute_scales_fun=compute_scales_fun,
     )
+
+
+def init_par(model, init_location: str, init_scale: str) -> Tuple[np.ndarray, np.ndarray, bool, bool]:
+    r"""
+    standard:
+    Only initialise intercept and keep other coefficients as zero.
+
+    closed-form:
+    Initialize with Maximum Likelihood / Maximum of Momentum estimators
+
+    Idea:
+    $$
+        \theta &= f(x) \\
+        \Rightarrow f^{-1}(\theta) &= x \\
+            &= (D \cdot D^{+}) \cdot x \\
+            &= D \cdot (D^{+} \cdot x) \\
+            &= D \cdot x' = f^{-1}(\theta)
+    $$
+    """
+    train_loc = False
+
+    groupwise_means = None
+
+    init_location_str = init_location.lower()
+    # Chose option if auto was chosen
+    auto_or_closed_form = init_location_str == "auto" or init_location_str == "closed_form"
+    if auto_or_closed_form or init_location_str == "all_zero":
+        if auto_or_closed_form:
+            logger.warning(
+                (
+                    "There is no need for closed form location model initialization"
+                    "because it is already closed form - falling back to zeros"
+                )
+            )
+        init_theta_location = np.zeros([model.num_loc_params, model.num_features])
+    elif init_location_str == "standard":
+        overall_means = np.mean(model.x, axis=0)  # directly calculate the mean
+        init_theta_location = np.zeros([model.num_loc_params, model.num_features])
+        init_theta_location[0, :] = np.log(overall_means)
```

Comment:
why is this log? Isn't the linker function the identity here? Because we are assume a normal distribution?

Reply:
yup.